syntax = "proto3";

package raft;

service Raft {
	rpc AppendEntries(AppendEntriesRequest) returns (AppendEntriesResponse) {}
	rpc RequestVote(RequestVoteRequest) returns (RequestVoteResponse) {}
}

message AppendEntriesRequest {
	int32 term = 1;
	int32 leaderId = 2;
	int32 prevLogIndex = 3;
	int32 prevLogTerm = 4;
	repeated string entries = 5;
	int32 leaderCommit = 6;
}
message AppendEntriesResponse {
	int32 term = 1;
	bool success = 2;
}

message RequestVoteRequest {
	int32 term = 1;
	int32 candidateId = 2;
	int32 lastLogIndex = 3;
	int32 lastLogTerm = 4;
}
message RequestVoteResponse {
	int32 term = 1;
	int32 voteGranted = 2;
}

message LogEntry {
	int32 term = 1;
	string command = 2;
}

message PersistentState {
  int32 currentTerm = 1;
  int32 votedFor = 2;
}
